### ISSUE TITLE
proxy-injector causes network bound init containers to fail

### ISSUE THREAD

**Issue by sjama** [linkerd/linkerd2#1760]
### What is the issue?
I was testing the auto-inject feature (edge-18.10.2). The auto-injection of the sidecar works as expected unless I specify an init container that touches the network.

This init container I specified tries to download a file from a gcs bucket but fails complaining about a network error, this causes the entire deployment to fail since init containers must run to completion for the deployment to be successful.

### How can it be reproduced?
Specify an init container on a deployment object that attempts to make a network call while using the auto-inject feature on edge-18.10.2

### Logs, error output, etc

```
ERROR: (gcloud.auth.activate-service-account)
This may be due to network connectivity issues. Please check your network settings, and the status of the service you are trying to reach.
```

#### `linkerd check` output

```
kubernetes-api: can initialize the client..................................[ok]
kubernetes-api: can query the Kubernetes API...............................[ok]
kubernetes-api: is running the minimum Kubernetes API version..............[ok]
linkerd-api: control plane namespace exists................................[ok]
linkerd-api: control plane pods are ready..................................[ok]
linkerd-api: can initialize the client.....................................[ok]
linkerd-api: can query the control plane API...............................[ok]
linkerd-api[kubernetes]: control plane can talk to Kubernetes..............[ok]
linkerd-api[prometheus]: control plane can talk to Prometheus..............[ok]
linkerd-version: can determine the latest version..........................[ok]
linkerd-version: cli is up-to-date.........................................[ok]
linkerd-version: control plane is up-to-date...............................[ok]

Status check results are [ok]
```

### Environment

- Kubernetes Version:

```
Client Version: v1.9.7
Server Version: v1.10.3
```

- Cluster Environment: (GKE, AKS, kops, ...)
```
Docker for Mac Version 18.06.1-ce-mac73 (26764)
```

- Host OS:
```
MacOs High Sierra Version 10.13.6
```

- Linkerd version:
```
Client version: edge-18.10.2
Server version: edge-18.10.2
```

### Possible solution
As suggested by @klingerf on slack the issue is likely caused by linkerd proxy rewriting the pods iptables before the proxy is fully operational. A possible solution is to ensure the linkerd auto-injection happens after init containers have all completed.

**Comment by ihcsim** [comment]
Can confirm that the proxy-injector puts the proxy-init container ahead of other init containers as seen [here](https://github.com/linkerd/linkerd2/blob/1fe19bf3ced627e05f03bec849434aac6d653732/controller/proxy-injector/patch.go#L10), while the CLI `inject` [appends it to the end of the list](https://github.com/linkerd/linkerd2/blob/1fe19bf3ced627e05f03bec849434aac6d653732/cli/cmd/inject.go#L362).

**Comment by klingerf** [comment]
This was fixed by #1763. Will go out in this week's release.
